webpack/components/TypeAhead/TypeAhead.js

```diff
@@ -9,81 +9,91 @@ import { default as TypeAheadSearchPf4 } from './pf4Search/TypeAheadSearch';
 import { getActiveItems } from './helpers/helpers';
 
 import './TypeAhead.scss';
+import useDebounce from '../../utils/useDebounce';
 
-class TypeAhead extends Component {
-  constructor(props) {
-    super(props);
+const TypeAhead = ({
+  items,
+  isDisabled,
+  onInputUpdate,
+  onSearch,
+  actionText,
+  initialInputValue,
+  patternfly4,
+  autoSearchEnabled,
+  autoSearchDelay,
+  bookmarkController,
+}) => {
+  const [inputValue, setInputValue] = useState(initialInputValue);
 
-    this.state = {
-      inputValue: this.props.initialInputValue,
-    };
-  }
+  const debouncedValue = useDebounce(inputValue, autoSearchDelay);
+  useEffect(
+    () => {
+      onInputUpdate(debouncedValue);
+    },
+    // This is needed to prevent cross-repo hook dependencies.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
`onInputUpdate` is defined in `Search.js` in Katello..

Reply:
Ahh yes, the dependencies here are optionally passed down from props to tableWrapper > onSearch > here. 
Removing this line.